--- menu141.ini	2011-01-03 23:20:32.000000000 +0100
+++ menu142.ini	2011-01-21 16:06:30.000000000 +0100
@@ -63,6 +63,10 @@
 ACCESS=crm_other
 module=crm/dokument.php
 
+[CRM--Zeiterfassung]
+ACCESS=crm_service
+module=crm/timetrack.php
+
 [CRM--Service]
 submenu=1
 
@@ -97,6 +101,10 @@
 ACCESS=crm_admin
 module=crm/tcatedit.php
 
+[CRM--Admin--Sonderflag]
+ACCESS=crm_admin
+module=crm/sflagedit.php
+
 [CRM--Admin--Mitteilungen]
 ACCESS=crm_admin
 module=crm/user3.php
