--- /tmp/menu.ini	2005-05-20 09:03:54.750257400 +0200
+++ /var/www/cusori/menu.ini	2005-04-28 12:31:08.000000000 +0200
@@ -21,6 +21,18 @@
 [CRM--Personen------erfassen]
 module=crm/personen3.php
 
+[CRM--Wartungsvertrag]
+module=crm/vertrag1.php
+
+[CRM--Wartungsvertrag------erfassen]
+module=crm/vertrag3.php
+
+[CRM--Maschinen]
+module=crm/maschine1.php
+
+[CRM--Maschinen------erfassen]
+module=crm/maschine3.php
+
 [CRM--Termine]
 module=crm/termin.php
 
